topics/tutorials/code_documentation.md

```diff
@@ -0,0 +1,33 @@
+[//]: # (title: Documentation)
+
+<!-- Copyright 2000-2022 JetBrains s.r.o. and other contributors. Use of this source code is governed by the Apache 2.0 license that can be found in the LICENSE file. -->
+
+[Quick Documentation](https://www.jetbrains.com/help/idea/viewing-reference-information.html#inline-quick-documentation)
+helps users by showing documentation, e.g., for classes, functions, or methods inside the editor.
+Plugin authors implement
+[`DocumentationProvider`](upsource:///platform/analysis-api/src/com/intellij/lang/documentation/DocumentationProvider.java)
+to show documentation for particular Psi elements.
+
+`DocumentationProvider` can be used for existing languages as well as for a custom language implementation.
+For an existing language, the implementation of `DocumentationProvider` is registered as `com.intellij.documentationProvider` extension point (EP),
```

Comment:
This difference is unclear to me. The questions that came to my mind:

- Can I register documentation provider via `com.intellij.documentationProvider` for my custom language?
- Can I register documentation provider via `com.intellij.lang.documentationProvider` for Java (existing language)?

If both answers are "yes", then this paragraph is a bit misleading IMHO.

Isn't it about performance?
So if we know the documentation language, we should register it via `com.intellij.lang.documentationProvider`. If our documentation provider can provide doc in various contexts, then we should use `com.intellij.documentationProvider`. Is it correct or I'm missing the point?

Reply:
1) you should use `lang` when targeting specific Language only
2) yes, it's possible to augment/override/.. existing DocProvider